### PR TITLE
fix: ensure reranking kwargs are properly passed

### LINKED ISSUE #1056: RerankingEvaluator._encode_unique_texts() got an unexpected keyword argument 'batch_size'
I was running an evaluation, and I think I found an issue with a recent change starting from v1.12.66

```2024-07-06 11:09:38.842489 >>> MindSmallReranking
Traceback (most recent call last):
  File "/home/lee/anaconda3/envs/wordllama/lib/python3.11/site-packages/mteb/evaluation/MTEB.py", line 382, in run
    results, tick, tock = self._run_eval(
                          ^^^^^^^^^^^^^^^
  File "/home/lee/anaconda3/envs/wordllama/lib/python3.11/site-packages/mteb/evaluation/MTEB.py", line 259, in _run_eval
    results = task.evaluate(
              ^^^^^^^^^^^^^^
  File "/home/lee/anaconda3/envs/wordllama/lib/python3.11/site-packages/mteb/abstasks/AbsTask.py", line 118, in evaluate
    scores[hf_subset] = self._evaluate_subset(
                        ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lee/anaconda3/envs/wordllama/lib/python3.11/site-packages/mteb/abstasks/AbsTaskReranking.py", line 40, in _evaluate_subset
    scores = evaluator(model)
             ^^^^^^^^^^^^^^^^
  File "/home/lee/anaconda3/envs/wordllama/lib/python3.11/site-packages/mteb/evaluation/evaluators/RerankingEvaluator.py", line 74, in __call__
    scores = self.compute_metrics(model)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lee/anaconda3/envs/wordllama/lib/python3.11/site-packages/mteb/evaluation/evaluators/RerankingEvaluator.py", line 79, in compute_metrics
    self.compute_metrics_batched(model)
  File "/home/lee/anaconda3/envs/wordllama/lib/python3.11/site-packages/mteb/evaluation/evaluators/RerankingEvaluator.py", line 115, in compute_metrics_batched
    all_query_embs = self._encode_unique_texts(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: RerankingEvaluator._encode_unique_texts() got an unexpected keyword argument 'batch_size'
```

After reverting to v1.12.65, the MindSmallReranking tasks completes successfully.

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Fixes #1056 

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/evaluation/evaluators/RerankingEvaluator.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def _encode_candidates_batched(
             all_docs,
             encode_corpus_func,
             prompt_name=self.task_name,
-            encode_kwargs=self.encode_kwargs,
+            **self.encode_kwargs,
```
Comment (bzantium) [inline mteb/evaluation/evaluators/RerankingEvaluator.py:213]:
```suggestion
            encode_kwargs=self.encode_kwargs,
```

File: mteb/evaluation/evaluators/RerankingEvaluator.py [hunk 2]
```diff
@@ -422,7 +422,7 @@ def _encode_unique_texts(
         all_texts: list[str],
         encode_fn: Callable,
         prompt_name: str | None,
-        encode_kwargs: dict[str, Any],
+        **encode_kwargs: Any,
```
Comment (bzantium) [inline mteb/evaluation/evaluators/RerankingEvaluator.py:425]:
```suggestion
        encode_kwargs: dict[str, Any],
```

**Review by bzantium — CHANGES_REQUESTED** [review]
Instead of make changes on two line, you can fix just one line following:
https://github.com/embeddings-benchmark/mteb/blob/061c3e3e02e2769430678580bb72765b84670807/mteb/evaluation/evaluators/RerankingEvaluator.py#L115-L120

into:
```
 all_query_embs = self._encode_unique_texts( 
     all_query_flattened, 
     encode_queries_func, 
     prompt_name=self.task_name, 
     encode_kwargs=self.encode_kwargs, 
 ) 
 ```

**Comment by KennethEnevoldsen** [comment]
> Instead of make changes on two line, you can fix just one line following:

Correct, this was just to unpack it for all encode functions (consistency across codebase). don't have a strong preference for either though. Assuming you don't either I will enable auto-merge here, but do feel free to update the PR to your suggestion and rerun the tests to ensure that it works.

**Pull request merged**